std/string.d

```diff
@@ -257,6 +257,16 @@ inout(wchar)[] fromStringz(inout(wchar)* cString) @nogc @system pure nothrow
     assert(fromStringz(foo) == null);
     assert(fromStringz("foo"w.ptr) == "foo"w);
     assert(fromStringz("foo\0"w.ptr) == "foo"w);
+    assert(fromStringz("福\0"w.ptr) == "福"w);
+}
+
+@system pure unittest
+{
+    // Test for all surrogate pairs
+    // (its a total of 1048576 itterations but its fast enough)
+    for (wchar i = 0xD800; i < 0xDC00; ++i)
+        for (wchar j = 0xDC00; j < 0xE000; ++j)
+            assert(fromStringz([j, i, i, j, '\u0000'].ptr) == [j, i, i, j]);
```

Comment:
FYI a reversed surrogate pair isn't valid UTF-16.

Reply:
UTF16 has no specified endianness, so while the sequence as a whole is never valid, it covers both little and big endian sequences.